cmd/gapit/screenshot.go

```diff
@@ -248,6 +254,45 @@ func (verb *screenshotVerb) frameCommands(ctx context.Context, capture *path.Cap
 	return commands, nil
 }
 
+func (verb *screenshotVerb) executedDrawCommands(ctx context.Context, capture *path.Capture, client client.Client, maxAmount int) ([]*path.Command, error) {
+	filter, err := verb.CommandFilterFlags.commandFilter(ctx, client, capture)
+	if err != nil {
+		return nil, log.Err(ctx, err, "Couldn't get filter")
+	}
+	filter.OnlyExecutedDraws = true
+
+	treePath := capture.CommandTree(filter)
+
+	boxedTree, err := client.Get(ctx, treePath.Path(), nil)
+	if err != nil {
+		return nil, log.Err(ctx, err, "Failed to load the command tree")
+	}
+
+	tree := boxedTree.(*service.CommandTree)
+
+	var allDrawCommands []*path.Command
+	traverseCommandTree(ctx, client, tree.Root, func(n *service.CommandTreeNode, prefix string) error {
+		if n.Group != "" || n.NumChildren > 0 {
+			return nil
+		}
+		allDrawCommands = append(allDrawCommands, n.Commands.First())
+		return nil
+	}, "", true)
+
+	if len(allDrawCommands) > maxAmount {
+		commands := make([]*path.Command, maxAmount)
+		step := float32(len(allDrawCommands) - 1) / float32(maxAmount - 1)
+	
+		for i := 0; i < len(commands); i++ {
+			commands[i] = allDrawCommands[int32(float32(i) * step)]
+		}
```

Comment:
Why the `-1`s  here? Also, this can lead to a division by zero if `maxAmount == 1`?

I think a simpler solution would be:

```
commands := []*path.Command{}
step := len(allDrawCommands) / maxAmount
if step == 0 {
  step = 1
}
for i := 0; i < len(allDrawCommands); i += step {
  commands = append(commands, allDrawCommands[i])
}
```

Reply:
-1 helps with making sure our evenly spaced indices ends with the last index. Otherwise, it will stop early. For example, 30/10 will stop at 27, but 29/19 will stop at 29. It's not a big deal. I just felt that it should be spaced between the first and last item.

Reply:
So. I did a quick test of your method. If len(allDrawCommands) == 5 and maxAmount == 3, then step == 1. Meaning, we would end up appending 5 commands, not 3.

Reply:
As discussed offline, let's keep the float step approach but remove the `-1`'s.